db/document.go

```diff
@@ -525,7 +528,8 @@ func (doc *Document) IsSGWrite(rawBody []byte) (isSGWrite bool, crc32Match bool)
 		base.Warnf("Unable to marshal doc body during SG write check for doc %s. Error: %v", base.UD(doc.ID), err)
 		return false, false
 	}
-	if base.Crc32cHashString(bodyBytes) == doc.SyncData.Crc32c {
+
+	if base.Crc32cHashString(bodyBytes) == doc.SyncData.Crc32c && emptyBodyCrc32cHash != doc.SyncData.Crc32c {
```

Comment:
This approach isn't valid - it doesn't handle the following case:
1. Empty document created
2. Non-body mutation made to empty document (eventing xattr, expiry, etc)

In this scenario, isSGWrite should be returning (true, true).  With the above clause, it's going return (false,false) and trigger an unwanted import.

Reply:
Changes are committed to handle this scenario in the latest revision.